packages/jazz-auth-clerk/src/index.ts

```diff
@@ -37,13 +38,47 @@ export class JazzClerkAuth {
     });
   }
 
-  onClerkUserChange = async (clerkClient: Pick<MinimalClerkClient, "user">) => {
-    if (!clerkClient.user) {
+  static async initializeAuth(clerk: MinimalClerkClient) {
+    const secretStorage = new AuthSecretStorage();
+
+    if (!isClerkCredentials(clerk.user?.unsafeMetadata)) {
       return;
     }
 
+    await JazzClerkAuth.loadClerkAuthData(
+      clerk.user.unsafeMetadata,
+      secretStorage,
+    );
+  }
+
+  private isFirstCall = true;
+
+  registerListener(clerkClient: MinimalClerkClient) {
+    let previousUser: MinimalClerkClient["user"] | null =
+      clerkClient.user ?? null;
+
+    // Need to use addListener because the clerk user object is not updated when the user logs in
+    return clerkClient.addListener((event) => {
+      const user = (event as Pick<MinimalClerkClient, "user">).user ?? null;
+
+      if (!isClerkAuthStateEqual(previousUser, user) || this.isFirstCall) {
+        this.onClerkUserChange({ user });
+        previousUser = user;
+        this.isFirstCall = false;
+      }
+    });
+  }
```

Comment:
Moved the logic from the framework adapters here to reduce duplicates and make it more testable.